desktop-widgets/tab-widgets/TabDiveInformation.cpp

```diff
@@ -20,6 +24,35 @@ TabDiveInformation::TabDiveInformation(QWidget *parent) : TabBase(parent), ui(ne
 	QStringList atmPressTypes { "mbar", get_depth_unit() ,"use dc"};
 	ui->atmPressType->insertItems(0, atmPressTypes);
 	pressTypeIndex = 0;
+	// This needs to be the same order as enum dive_comp_type in dive.h!
+	QStringList types = QStringList();
+	for (int i = 0; i < NUM_DIVEMODE; i++)
+		types.append(gettextFromC::tr(divemode_text_ui[i]));
+	ui->diveType->insertItems(0, types);
+	connect(ui->diveType, SIGNAL(currentIndexChanged(int)), this, SLOT(diveMode_Changed(int)));
+	ui->scrollAreaWidgetContents_3->setStyleSheet("QGroupBox { border: 1px solid #c0c0c0; border-radius: 4px; margin-top: 0.65em; background-color: #e4e4e4;} QGroupBox::title { color: #0000c0;} ");
+	ui->diveModeBox->setStyleSheet("QGroupBox{ padding: 0;} ");
+	ui->diveHeadingLabel->setStyleSheet(CSS_SET_HEADING_BLUE);
+	ui->gasHeadingLabel->setStyleSheet(CSS_SET_HEADING_BLUE);
+	ui->environmentHeadingLabel->setStyleSheet(CSS_SET_HEADING_BLUE);
+	ui->groupBox_surge->setStyleSheet(CSS_SET_SMALL_LABEL);
+	ui->groupBox_visibility->setStyleSheet(CSS_SET_SMALL_LABEL);
+	ui->groupBox_chill->setStyleSheet(CSS_SET_SMALL_LABEL);
+	ui->groupBox_waves->setStyleSheet(CSS_SET_SMALL_LABEL);
+	ui->groupBox_current->setStyleSheet(CSS_SET_SMALL_LABEL);
+	QAction *action = new QAction(tr("OK"), this);
+	connect(action, &QAction::triggered, this, &TabDiveInformation::closeWarning);
+	ui->multiDiveWarningMessage->addAction(action);
+	action = new QAction(tr("Undo"), this);
+	connect(action, &QAction::triggered, Command::undoAction(this), &QAction::trigger);
+	connect(action, &QAction::triggered, this, &TabDiveInformation::closeWarning);
+	ui->multiDiveWarningMessage->addAction(action);
+	ui->multiDiveWarningMessage->hide();
+//	ui->groupBox_waves->hide();
+//	ui->groupBox_visibility->hide();
+//	ui->groupBox_current->hide();
+//	ui->groupBox_surge->hide();
+//	ui->groupBox_chill->hide();
 }
```

Comment:
adding commented out code seems suboptimal (sometimes there's a good reason to do this - here I don't see that)

Reply:
Done.